packages/smart-contracts/packages/nft/test/unit/Nft/mintAndUpdateNftData.t.sol

```diff
@@ -152,11 +152,58 @@ contract Nft_MintAndUpdateNftData_Unit_Test is Unit_Test {
         assertEq(nftData.extendedActivityEndTimestamp, 0);
     }
 
+    function test_mintAndUpdateNftData_DeletesPreviousNft()
+        external
+        mintLevel2NftForAlice
+        mintLevel2NftForAlice
+    {
+        vm.prank(admin);
+        nft.mintAndUpdateNftData(alice, NFT_TOKEN_ID_1, LEVEL_3);
+
+        assertEq(nft.getActiveNft(alice), 0, "Active NFT ID is incorrect");
+    }
+
```

Comment:
Nft is not deleted here as `activeNft` is never assigned. Activate first to see that active nft is deleted

Reply:
Or change naming, as I am not sure what you wanted to test in this unit test.